app/models/message_object.rb

```diff
@@ -135,4 +136,12 @@ def unassign_tag(tag)
   def thread
     message.thread
   end
+
+  def remove_object_related_tags_from_thread
+    tags.each do |tag|
+      message.thread.unassign_tag(tag) unless message.thread.objects.where.not(id: id).map(&:tags).flatten&.map(&:id).include?(tag.id)
+    end
```

Comment:
Toto sa pomerne zle cita, znamena tato logika, ze mazeme vsetky tagy take kde uz neexistuje taky tag na nejakom inom objekte?

Ak ano tak to skusme zabalit do nejakej tak nazvanej metody.

Reply:
Ano, tak. Upracem.